presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analyzer.java

```diff
@@ -75,14 +75,28 @@ public Analysis analyze(Statement statement)
     }
 
     public Analysis analyze(Statement statement, boolean isDescribe)
+    {
+        Analysis analysis = analyzeSemnatic(statement, isDescribe);
+        aclCheck(analysis);
+        return analysis;
+    }
+
+    public Analysis analyzeSemnatic(Statement statement, boolean isDescribe)
     {
         Statement rewrittenStatement = StatementRewrite.rewrite(session, metadata, sqlParser, queryExplainer, statement, parameters, accessControl, warningCollector);
         Analysis analysis = new Analysis(rewrittenStatement, parameters, isDescribe);
         StatementAnalyzer analyzer = new StatementAnalyzer(analysis, metadata, sqlParser, accessControl, session, warningCollector);
         analyzer.analyze(rewrittenStatement, Optional.empty());
         analyzeForUtilizedColumns(analysis, analysis.getStatement());
+        return analysis;
+    }
 
-        // check column access permissions for each table
+    /**
+     * check column access permissions for each table
```

Comment:
`AccessControl` has a lot of method to check ACL at different places. Not quite sure if we call it ACL checking here is precise. Will leave it to @shixuan-fan for further design decision making :D

Reply:
This is a good point, we should probably call this `checkColumnAccessPermissions` :)

Reply:
> This is a good point, we should probably call this `checkColumnAccessPermissions` :)

Thanks! Good to know. I think I'll rename the Scuba column as well to this name .